src/sidebars/search/search-controller.js

```diff
@@ -62,8 +62,12 @@
 
                 $scope.$on('updateFavourites', function(event, data) {
                     var index = self.stocks.map(function(stock) { return stock.code; }).indexOf(data.code);
-                    if (index > -1 && !self.query) {
-                        self.stocks.splice(index, 1);
+                    if (index > -1) {
+                        if (!self.query) {
+                            self.stocks.splice(index, 1);
+                        } else {
+                            self.stocks[index].favourite = false;
+                        }
```

Comment:
Can we have some code comment here? What this block does? I know there was no comment before, but it is becoming bit complicated.